### PR TITLE
Ensure http_response_header is a list<string>

### PR DESCRIPTION
**Pull request by Seldaek**
Not sure if that's the proper/best use of AccessoryArrayListType, but anyway I hope the intent is clear :)

### CODE REVIEW

**Comment by staabm** [comment]
since no test is failling after the change it seems we are missing a NodeScopeResolverTest with a type inference assertion for this variable :)

**Review by staabm — APPROVED** [review]
lgtm - thank you

File: tests/PHPStan/Analyser/data/http-response-header.php [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<?php
+
+use function PHPStan\Testing\assertType;
+
+file_get_contents('https://example.org');
+assertType('list<string>', $http_response_header);
```
Comment (staabm) [inline tests/PHPStan/Analyser/data/http-response-header.php:7]: ohh I think you are missing another assert for `assertNativeType`

Comment (Seldaek) [reply tests/PHPStan/Analyser/data/http-response-header.php:7]: Ok hopefully I did this correctly :)

**Comment by ondrejmirtes** [comment]
Thank you!

**Pull request merged**